server/storage/backend/batch_tx_test.go

```diff
@@ -205,3 +207,62 @@ func TestBatchTxBatchLimitCommit(t *testing.T) {
 		return nil
 	})
 }
+
+func TestRangeAfterDeleteMatch(t *testing.T) {
+	b, _ := betesting.NewTmpBackend(t, time.Hour, 10000)
+	defer betesting.Close(t, b)
+
+	tx := b.BatchTx()
+
+	tx.Lock()
+	tx.UnsafeCreateBucket(schema.Test)
+	tx.UnsafePut(schema.Test, []byte("foo"), []byte("bar"))
+	tx.Unlock()
+	tx.Commit()
+
+	checkTxnResponseMatch(t, b.BatchTx(), b.ReadTx(), []byte("foo"), nil, 0)
+
+	tx.Lock()
+	tx.UnsafeDelete(schema.Test, []byte("foo"))
```

Comment:
This is incorrect usage. `tx.UnsafeDelete` shouldn't be called directly to remove keys from the `key` bucket. Instead, etcd still puts a key/value with tombstone. The only exception is compaction.

Reply:
Just ignore this comment, it makes sense to call UnsafeDelete on other buckets.